app/queries/available_user_units_query.rb

```diff
@@ -0,0 +1,20 @@
+class AvailableUserUnitsQuery
+  def initialize(user:)
+    @user = user
+    @units = []
+  end
+
+  def to_units_array
+    @user.units.each do |unit|
```

Comment:
```suggestion
    @user.units.flat_map do |unit|
```

Взагалі зі змінною `@units` тут якийсь месс.  Думаю її тут непотрібно взагалі.